extensions/desktop-common/desktop-init

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -e
+
+#################
+# Launcher init #
+#################
+
+# On Fedora $SNAP is under /var and there is some magic to map it to /snap.
+# # We need to handle that case and reset $SNAP
+SNAP="${SNAP/\/var\/lib\/snapd/}"
```

Comment:
This seems very strange on first sight. There are reasons why $SNAP points elsewhere in Fedora, and those reasons are trusted by snaps in general. That change breaks the common interface of snapd itself, and can easily break down other things in the system. On the other hand, internally in the snap, /snap still exists, and $SNAP should point to /snap _already_ if it's not a classic snap, so if that's not the case today we have a bug in snapd. If it is a classic snap, then it points elsewhere because /snap may not exist, which means there's not a good reason for it to point there anyway.

Can someone please clarify these issues with the snapd team?

Reply:
This bit of code is from the [ubuntu/snapcraft-desktop-helpers](https://github.com/ubuntu/snapcraft-desktop-helpers) repo. I don't know why it's needed but I also don't know for certain that it isn't, so I left it in place.

Reply:
If we are going to err, let's err on the safe side and drop this logic. Per rationale above, this is a fundamental setting made available to every snap by snapd. It shouldn't be hacked like this by first-class command chain scripts.

Reply:
changeset fd9cdd9 removes this.